### PR TITLE
[SofaCommon] Pluginize SofaImplicitOdeSolver

### PR DESCRIPTION
**Pull request by fredroy**
#1597

- Usual cleaning (pragma, namespaces)
- cmake dep polish
- remove useless includes

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaExplicitOdeSolver/src/SofaExplicitOdeSolver/initSofaExplicitOdeSolver.h [hunk 1]
```diff
@@ -25,4 +25,4 @@
 namespace sofa::component
 {
 
-} // namespace sofa::component
+} // namespace sofa::component
```
Comment (epernod) [inline SofaKernel/modules/SofaExplicitOdeSolver/src/SofaExplicitOdeSolver/initSofaExplicitOdeSolver.h]: no new line at end

File: SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp [hunk 2]
```diff
@@ -471,7 +475,7 @@ std::map<std::string, ComponentChange> uncreatableComponents = {
     { "TriangleSetTopologyAlgorithms", Removed("v20.12", "v20.12") },
     { "EdgeSetTopologyAlgorithms", Removed("v20.12", "v20.12") },
     { "PointSetTopologyAlgorithms", Removed("v20.12", "v20.12") },
-
+
```
Comment (epernod) [inline SofaKernel/modules/SofaHelper/src/sofa/helper/ComponentChange.cpp]: indent artifact

File: SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/BarycentricPenalityContact.h [hunk 3]
```diff
@@ -136,4 +136,4 @@ extern template class SOFA_SOFAMESHCOLLISION_API BarycentricPenalityContact<Cyli
 extern template class SOFA_SOFAMESHCOLLISION_API BarycentricPenalityContact<CylinderCollisionModel<sofa::defaulttype::Rigid3Types>, OBBCollisionModel<sofa::defaulttype::Rigid3Types>>;
 #endif
 
-} // namespace sofa::component::collision
+} //namespace sofa::component::collision
```
Comment (epernod) [inline SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/BarycentricPenalityContact.h:139]: space removed not needed

Comment (fredroy) [reply SofaKernel/modules/SofaMeshCollision/src/SofaMeshCollision/BarycentricPenalityContact.h:139]: this comes from a script and this is not the only one affected... a bit lazy to correct all 🤐

**Pull request merged**